code/modules/food/food/snacks.dm

```diff
@@ -39,12 +39,20 @@
 	if(nutriment_amt)
 		reagents.add_reagent("nutriment",(nutriment_amt*2),nutriment_desc)		//VOREStation Edit: Undoes global nutrition nerf
 
-	//Placeholder for effect that trigger on eating that aren't tied to reagents.
+//Placeholder for effect that trigger on eating that aren't tied to reagents.
 /obj/item/weapon/reagent_containers/food/snacks/proc/On_Consume(var/mob/M)
 	if(!usr)
 		usr = M
 	if(!reagents.total_volume)
 		M.visible_message("<span class='notice'>[M] finishes eating \the [src].</span>","<span class='notice'>You finish eating \the [src].</span>")
+		// Embedded-in-food smol vore
+		for(var/obj/item/weapon/holder/holder in src)
+			if(holder.held_mob?.devourable && isliving(M))
```

Comment:
This is redundant for several reasons:
- `holder.held_mob` is defined as `var/mob/living/held_mob` on holder
- `var/devourable` is defined on `/mob/living`
You can just check `if(holder.held_mob?.devourable)` although if held_mob is null that implies something has gone terribly wrong somewhere and an invalid holder hasn't been qdel'd.

Reply:
I think you may have misread. The check is `isliving(M)`. `M` is the mob eating the food, no relation to holder.

Reply:
Aaaaughhhhh yeah I misread. `On_Consume()` on snacks is only ever called for carbons anyway, though, so it's of dubious veracity. Could probably just change it to `On_Consume(var/mob/living/carbon/C)`.
Oh also if you do keep the isliving check please move it OUT of the loop, it only needs to be called once.

Reply:
True about the last part! Where's the only call? I want to go look

Reply:
There's also snacks.dm:290 where /mob/living can eat snacks by attacking them, so it's not only carbons.

This whole system is dumm. Should probably just give mobs an attack_food() or somesuch